### PR TITLE
[APM-CI] Jenkins Pipeline for Agent Node.js

### PR DESCRIPTION
**Pull request by kuisathaverat**
Jenkins file to build the APM Agent Node.js with the multibranch pipeline job https://apm-ci.elastic.co/job/apm-agent-nodejs/job/apm-agent-nodejs-mbp/

image

### CODE REVIEW

**Comment by kuisathaverat** [comment]
Tests results are loaded

screenshot 2019-01-22 at 16 47 47

File: test/script/docker/run_tests.sh [hunk 1]
```diff
@@ -24,17 +24,17 @@ else
...
-  -t apm-agent-nodejs:node-${1} \
-  ./test
-
 NODE_VERSION=${1} docker-compose -f ./test/docker-compose.yml -f ./test/docker-compose.ci.yml run \
-  -e NODE_VERSION=${NODE_VERSION} -e TAV=${TAV_VERSIONS} -e CI=true \
+  -e NODE_VERSION=${NODE_VERSION} \
+  -e TAV=${TAV_VERSIONS} \
+  -e JUNIT=${JUNIT} \
+  -e CI=true \
+  -e ELASTIC_APM_ASYNC_HOOKS=false \
```
Comment (Qard) [inline test/script/docker/run_tests.sh]: Why are we disabling async_hooks here?

Comment (kuisathaverat) [reply test/script/docker/run_tests.sh]: C&P from Travis file

Comment (Qard) [reply test/script/docker/run_tests.sh]: Ah, we _should_ actually be testing both states. On Travis, some jobs include that and some don't. Specifically, every version of Node.js from 8 and above should test both states.

Comment (Qard) [reply test/script/docker/run_tests.sh]: The current setup tests with it enabled, so should we maybe just remove this line for now and land, then we can figure out a matrix test to cover both states later?

Comment (kuisathaverat) [reply test/script/docker/run_tests.sh]: yep, we can make it on the matrix test

**Comment by Qard** [comment]
I hope you don't mind, I've squashed and rebased so I can get a new test run that passes the commit lint test.

**Comment by kuisathaverat** [comment]
@Qard it is fine, seems like Node.js test does not end ever, so the job is killed after an hour, we have a timeout of one hour set.

**Comment by Qard** [comment]
Yeah, I've been looking into the pause. Not sure what the issue is yet. I've so far been unable to reproduce locally, with the same Node.js and module versions. 🤔

**Comment by kuisathaverat** [comment]
> Yeah, I've been looking into the pause. Not sure what the issue is yet. I've so far been unable to reproduce locally, with the same Node.js and module versions. 🤔

Could it be related to the memory amount on Jenkins agents? IIRC Jenkins Agents has about 4G of memory.

**Comment by kuisathaverat** [comment]
@alvarolobato I tested it yesterday, it seems something that it is fixed right now, not related to the CI, I have launched it again to be sure, if the CI pass the PR is ready to merge.

**Comment by kuisathaverat** [comment]
There is something on the test that sometimes fails with Node.js 6/6.0, the behavior it is that is stuck in some place, it is not related to CI.

**Pull request merged**